worlds/lingo/__init__.py

```diff
@@ -0,0 +1,149 @@
+    """
+    game = "Lingo"
+    web = LingoWebWorld()
+
+    base_id = 444400
+    topology_present = True
+    data_version = 1
+
+    static_logic = StaticLingoLogic()
+    static_items = StaticLingoItems(base_id)
+    static_locat = StaticLingoLocations(base_id)
+    option_definitions = lingo_options
+
+    item_name_to_id = {
+        name: data.code for name, data in static_items.ALL_ITEM_TABLE.items()
+    }
+    location_name_to_id = {
+        name: data.code for name, data in static_locat.ALL_LOCATION_TABLE.items()
+        if data.code is not None
+    }
+
+    # This is just used for unit testing. It should remain at the default values for actual play.
+    test_options: LingoTestOptions = LingoTestOptions()
+
+    def _get_slot_data(self):
+        return {
+            'door_ids_by_item_id': {
+                data.code: data.door_ids for data in self.static_items.ALL_ITEM_TABLE.values()
+                if data.code is not None and len(data.door_ids) > 0
+            },
+            'painting_ids_by_item_id': {
+                data.code: data.painting_ids for data in self.static_items.ALL_ITEM_TABLE.values()
+                if data.code is not None and len(data.painting_ids) > 0
+            },
+            'panel_ids_by_location_id': {
+                data.code: data.panel_ids() for data in self.static_locat.ALL_LOCATION_TABLE.values()
+                if data.code is not None
+            },
+            'seed': self.multiworld.per_slot_randoms[self.player].randint(0, 1000000),
+        }
+
+    def generate_early(self):
+        self.player_logic = LingoPlayerLogic(self.multiworld, self.player, self.static_logic, self.test_options)
+
+    def create_regions(self):
+        create_regions(self.multiworld, self.player, self.static_logic, self.player_logic)
+
+    def create_items(self):
+        pool = []
+
+        for name in self.player_logic.REAL_ITEMS:
+            new_item = self.create_item(name)
+            pool.append(new_item)
+
+        for location, item in self.player_logic.EVENT_LOC_TO_ITEM.items():
+            event_item = LingoItem(item, ItemClassification.progression, None, player=self.player)
+            location_obj = self.multiworld.get_location(location, self.player)
+            location_obj.place_locked_item(event_item)
+
+        if self.player_logic.FORCED_GOOD_ITEM != "":
+            new_item = self.create_item(self.player_logic.FORCED_GOOD_ITEM)
+            location_obj = self.multiworld.get_location("Starting Room - HI", self.player)
+            location_obj.place_locked_item(new_item)
+
+        item_difference = len(self.player_logic.REAL_LOCATIONS) - len(pool)
+        if item_difference > 0:
+            trap_percentage = get_option_value(self.multiworld, self.player, "trap_percentage")
+            traps = floor(item_difference * trap_percentage / 100.0)
+            non_traps = item_difference - traps
+
+            for i in range(0, non_traps):
```

Comment:
I'm not familiar with the game, but are there no flavor items you could add? Generally players don't like finding "Nothing"s. Not a blocker by any means though.
```suggestion
            pool += [self.create_item("Nothing") for i in range(non_traps)]
```